miio/devicestatus.py

```diff
@@ -196,17 +187,17 @@ def setting(
     """
 
     def decorator_setting(func):
-        property_name = func.__name__
+        property_name = str(func.__name__)
+        qualified_name = str(func.__qualname__)
 
-        if setter is None and setter_name is None:
-            raise Exception("Either setter or setter_name needs to be defined")
+        if setter_name is None:
+            raise Exception("Setter_name needs to be defined")
```

Comment:
Why is this parameter being removed?

Reply:
Since it will not work to put in a function directly in the decorator since the class object has not yet been created. Therefore always the setter_name schould be used and the binding of the setter methods schould take place when the class object gets created.

I tried to use the setter parameter in my code directly in the decorator, but I could not get it to work (I think it will never work).

Reply:
One way would be to give it a reference to a callable (like `DeviceClass.set_something`), and binding that to the instance object much like what is done for the `setter_name` one, no?`.

But that is not important at the moment, let's however keep the code in place just in case for the time being.

Reply:
I don't think that works since how would you get the instance object from the `DeviceClass.set_something` callable. But I am quite new to all these decorators, still figuring it all out a bit.
Learning a lot from your coding skills!

Schould I just keep the setter parameter and add a Exception that it is not yet implemented instead?

Reply:
Yes, just add an exception there. I'm also learning, so it's all good :-)

I think something like this could work for the binding, but I haven't tried it out:
```
from functools import partial
method = partial(setter, self)  # now the `self` is passed as the first parameter to the method calls
```

Reply:
I made the exception in a seperate PR.